Sources/Workspace/ToolsVersionSpecificationPrepender.swift

```diff
@@ -0,0 +1,92 @@
+
+/// Prepends a Swift tools version specification to the non-version-specific manifest file (`Package.swift`) in the given directory.
+///
+/// If the main manifest file already contains a valid tools version specification (ignoring the validity of the version specifier and that of everything following it), then the existing specification is replaced by this new one.
+///
+/// The version specifier in the specification does not contain any build metadata or pre-release identifier. The patch version is included if and only if it's not zero.
+///
+/// A `FileSystemError` is thrown if the manifest file is unable to be read from or written to.
+///
+/// - Precondition: `manifestDirectoryPath` must be a valid path to a directory that contains a `Package.swift` file.
+///
+/// - Parameters:
+///   - manifestDirectoryPath: The absolute path to the given directory.
+///   - toolsVersion: The Swift tools version to specify as the lowest supported version.
+///   - fileSystem: The filesystem to read/write the manifest file on.
+///
+/// - Throws: A `FileSystemError` instance, if the manifest file is unable to be located, read from, or written to..
+public func prependToolsVersionSpecification(toDefaultManifestIn manifestDirectoryPath: AbsolutePath, specifying toolsVersion: ToolsVersion, fileSystem: FileSystem) throws {
+    let manifestFilePath = manifestDirectoryPath.appending(component: Manifest.filename)
+    try prependToolsVersionSpecification(toManifestAt: manifestFilePath, specifying: toolsVersion, fileSystem: fileSystem)
+}
+
+// FIXME: Throw an error if the specified version is greater than the version-specific manifest's version?
+// For example, if the manifest file is Package@swift-4.0.swift and the given version to specify is 5.0.
+/// Prepends a Swift tools version specification to the specified manifest file.
+///
+/// If the main manifest file already contains a valid tools version specification (ignoring the validity of the version specifier and that of everything following it), then the existing specification is replaced by this new one.
+///
+/// The version specifier in the specification does not contain any build metadata or pre-release identifier. The patch version is included if and only if it's not zero.
+///
+/// A `FileSystemError` is thrown if the manifest file is unable to be located, read from, or written to.
+///
+/// - Precondition: `manifestFilePath` must be a valid path to a file.
+///
+/// - Parameters:
+///   - manifestFilePath: The absolute path to the specified manifest file.
+///   - toolsVersion: The Swift tools version to specify as the lowest supported version.
+///   - fileSystem: The filesystem to read/write the manifest file on.
+///
+/// - Throws: A `FileSystemError` instance, if the manifest file is unable to be read from or written to..
+public func prependToolsVersionSpecification(toManifestAt manifestFilePath: AbsolutePath, specifying toolsVersion: ToolsVersion, fileSystem: FileSystem) throws {
```

Comment:
This function is more general than `prependToolsVersionSpecification(toDefaultManifestIn:specifying:fileSystem:)`. It is able to write to version-specific manifest files. Should the function verify that the version to specify is no greater than the version in the manifest's file name? For example, reject using `6.0` as the version specifier if the manifest file is `Package@swift-5.0.swift`?

Reply:
Good question.  It kind of feels to me as if this function does a bit too much by generating the content as well as modifying the file in the file system.  Updating the tools version usually also involves changes to the manifest, so it doesn't feel like something that can be done in isolation.

Reply:
Sorry for this late response. What you wrote makes sense. 

Do you think it would be better if I just remove this function and the ability to write to version-specific manifests? Or, would it be better if I further factor this function to give each piece less responsibility?